modules/optimization/src/userobjects/DensityUpdateTwoConstraints.C

```diff
@@ -0,0 +1,292 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DensityUpdateTwoConstraints.h"
```

Comment:
A future refactor should try to create a base class to share common functions with DensityUpdate.

Reply:
Yes. That is a byproduct of not wanting to touch Max's initial implementation. That can be future work.